Content.Server/DeviceNetwork/Systems/DeviceNetworkSystem.cs

```diff
@@ -241,46 +294,66 @@ public sealed class BeforePacketSentEvent : CancellableEntityEventArgs
         /// <summary>
         /// The EntityUid of the entity the packet was sent from.
         /// </summary>
-        public EntityUid Sender;
+        public readonly EntityUid Sender;
+
+        public readonly TransformComponent SenderTransform;
+
+        /// <summary>
+        ///     The senders current position in world coordinates.
+        /// </summary>
+        public readonly Vector2 SenderPosition;
```

Comment:
My plan is to raise an event on the sending entity and letting systems modify the payload before sending and adding the payload to the BeforePacketSentEvent so you can have components that act on the payload. 
Having the transform and position retrieved inside the DeviceNetSystem and added to the event feels weird considering that but I can see why you did it.

Reply:
Yeah, its just that ever grid or wireless network recipient always retrieved the component individually which probably slowed things down a bit. But I'm, not sure why that's at odds with allow systems/components to modify the payload before it gets sent.

Reply:
Oh it's not at odds, just that in the future that kind of data can be added to the payload by components like the wireless component.
That's why it felt a bit weird to me at first but what you did is good 👌